test/test_c10d.py

```diff
@@ -3171,9 +3171,12 @@ def _test_nccl_errors_blocking(self, func):
         process_group.allreduce(torch.rand(10).cuda(self.rank))
         if self.rank == 0:
             work = process_group.allreduce(torch.rand(10).cuda(self.rank))
-            with self.assertRaises(RuntimeError):
+            try:
                 # Operation would time out in blocking mode.
                 work.wait()
+            except Exception as e:
```

Comment:
nit: are we not able to do self.assertRaisesRegex and keep the more  specific exception type?

Reply:
I'm intentionally try exception here to run a cuda operation after catching the exception, so that we can reproduce the nccl stuck issue (without fixing in this diff)